WIP: Cpp Import/Export

patrick-luethi opened the pull request:
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/dimitar-asenov/Envision/pull/300%23issuecomment-176288944%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/dimitar-asenov/Envision/pull/300%23issuecomment-176288944%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222016-01-28T17%3A16%3A02Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6049349%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/patrick-luethi%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-General Comment'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/dimitar-asenov/Envision/pull/300#issuecomment-176288944'>General Comment</a></b>
- <a href='https://github.com/patrick-luethi'><img border=0 src='https://avatars.githubusercontent.com/u/6049349?v=3' height=16 width=16'></a> done

<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/300?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/dimitar-asenov/Envision/pull/300?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/dimitar-asenov/Envision/pull/300'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

patrick-luethi committed:
ignore dependencies to inner declarations

this change aims to remove forward declarations and inclusions for inner declarations

patrick-luethi committed:
make cast types always introduce a hard dependency

patrick-luethi committed:
expand Q_DECLARE_FLAGS

this change allows us to resolve dependencies to the generated declaration

patrick-luethi committed:
do not topologically order fields inside declarations

field order should be determined by the programmer (memory layout).
therefore we do not order the fields and instead print them at the very end.

patrick-luethi committed:
import access modifier for variable declarations

patrick-luethi committed:
reconstruct eliminated prefix for template instantiation special cases

patrick-luethi committed:
ignore only dependencies on inner classes

fixed a bug where dependencies of references targeting a formal argument were ignored

dimitar-asenov commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/dimitar-asenov/Envision/pull/300'>Refresh</a>.<a href='#crh-update-none'></a>

patrick-luethi committed:
check for redundant forced dependency constructs

assert that constructs of the form (void)(Type*)nullptr only occur if we did not find a hard dependency on Type yet.

patrick-luethi committed:
make forced dependency construct detection faster and more specific

patrick-luethi committed:
elaborate comment on forced dependency assertion

patrick-luethi commented:
done
